packages/connect/src/ExtensionProvider/ExtensionProvider.ts

```diff
@@ -95,44 +95,24 @@ export class ExtensionProvider implements ProviderInterface {
     relayChain: string,
     parachain?: string,
   ) {
-    this.#chainId = nextChainId++
-    this.#appName = displayName
-
     /**
      * TODO: we should remove the chainName from the payload of the messages,
      * since this is information that doesn't have to be sent on every message and
      * the Extension can extract it from the chainSpecs, also that way we avoid
      * parsing a large JSON on the main thread.
      */
-    this.#chainName = JSON.parse(relayChain).name
     this.#chainSpecs = relayChain
     this.#connectionStatePingerId = null
     this.#parachainSpecs = ""
     if (parachain) {
       this.#parachainSpecs = parachain
     }
-  }
-
-  /**
-   * name
-   *
-   * @returns the name of this app to be used by the extension for display
-   * purposes.
-   *
-   * @remarks Apps are expected to make efforts to make this name reasonably
-   * unique.
-   */
-  public get name(): string {
-    return this.#appName
-  }
-
-  /**
-   * chainSpecs
-   *
-   * @returns the name of the chain this `ExtensionProvider` is talking to.
-   */
-  public get chainSpecs(): string {
-    return this.#chainSpecs
+    this.#commonMessageData = {
+      appName: displayName,
+      chainId: nextChainId++,
+      chainName: JSON.parse(relayChain).name,
+      origin: EXTENSION_PROVIDER_ORIGIN,
+    }
```

Comment:
name may not exist on the spec. This could do:
```suggestion
    const { name, id } = JSON.parse(relayChain)
    this.#commonMessageData = {
      appName: displayName,
      chainId: nextChainId++,
      chainName: name || id
      origin: EXTENSION_PROVIDER_ORIGIN,
    }
```

Reply:
In a future PR I will get rid of that `chainName`, as it shouldn't be passed at all in these messages... So, for now I don't want to bother improving something that we are about to get rid off.

Reply:
This PR doesn't change anything here, it is just copy-pasting the existing code. Fixing this is out of scope of this PR here, and will be fixed later in #572.